src/resource_clients/run.ts

```diff
@@ -279,24 +279,49 @@ export class RunClient extends ResourceClient {
             return undefined;
         }
         if (toLog === undefined || toLog === 'default') {
-            // Create default StreamedLog
-            // Get actor name and run id
-            const runData = await this.get();
-            const runId = runData?.id ?? '';
-
-            const actorId = runData?.actId ?? '';
-            const actorData = (await this.apifyClient.actor(actorId).get()) || { name: '' };
+            toLog = await this.getActorRedirectLog();
+        }
 
-            const actorName = actorData?.name ?? '';
-            const name = [actorName, `runId:${runId}`].filter(Boolean).join(' ');
+        return new StreamedLog({ logClient: this.log(), toLog, fromStart });
+    }
 
-            toLog = new Log({ level: LEVELS.DEBUG, prefix: `${name} -> `, logger: new LoggerActorRedirect() });
+    /**
+     * Get StatusMessageWatcher for convenient streaming of the Actor run status message and its redirection.
+     */
+    async getStatusMessageWatcher(
+        options: getStatusMessageWatcherOptions = {},
+    ): Promise<StatusMessageWatcher | undefined> {
+        let { toLog } = options;
+        if (toLog === null || !isNode()) {
+            // Explicitly no logging or not in Node.js
+            return undefined;
         }
+        if (toLog === undefined || toLog === 'default') {
+            toLog = await this.getActorRedirectLog();
+        }
+        return new StatusMessageWatcher({ toLog, runClient: this, checkPeriod: options.checkPeriod });
+    }
 
-        return new StreamedLog({ logClient: this.log(), toLog, fromStart });
+    private async getActorRedirectLog(): Promise<Log> {
+        // Get actor name and run id
+        const runData = await this.get();
+        const runId = runData ? `${runData.id ?? ''}` : '';
+
+        const actorId = runData?.actId ?? '';
+        const actorData = (await this.apifyClient.actor(actorId).get()) || { name: '' };
+
+        const actorName = runData ? (actorData.name ?? '') : '';
+        const name = [actorName, `runId:${runId}`].filter(Boolean).join(' ');
+
+        return new Log({ level: LEVELS.DEBUG, prefix: `${name} -> `, logger: new LoggerActorRedirect() });
```

Comment:
```suggestion
        const runData = await this.get();
        const runId = runData?.id ?? '';

        const actorId = runData?.actId ?? '';
        const actorData = (await this.apifyClient.actor(actorId).get()) || { name: '' };

        const actorName = actorData?.name ?? '';
        const name = [actorName, `runId:${runId}`].filter(Boolean).join(' ');

        return new Log({ level: LEVELS.DEBUG, prefix: `${name} -> `, logger: new LoggerActorRedirect() });
```

This is the removed code from L282, maybe this PR was made before merging the final version of log streaming?